src/functions/functions.ts

```diff
@@ -0,0 +1,45 @@
+import { FirebaseFunctions, HttpsCallableResult } from '@firebase/functions-types';
+import { FirebaseOptions } from '@firebase/app-types';
+import { Injectable, Inject, Optional, NgZone } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { FirebaseAppConfig, FirebaseAppName, _firebaseAppFactory, FirebaseZoneScheduler } from 'angularfire2';
+
+import 'rxjs/add/observable/fromPromise';
+import 'rxjs/add/operator/map'
+
+@Injectable()
+export class AngularFireFunctions {
+
+  /**
+   * Firebase Functions instance
+   */
+  public readonly functions: FirebaseFunctions;
+
+  public readonly scheduler: FirebaseZoneScheduler;
+
+  public httpsCallable<T, R>(name: string) {
+    const callable = this.functions.httpsCallable(name);
+    return (data: T) => {
+      return this.scheduler.runOutsideAngular(
+        Observable.fromPromise(callable(data))
+          .map(r => r.data as R)
```

Comment:
@davideast what do you think about mapping `.data` here? This is the only thing I'm on the fence about. It's nice ergo buuut what if the CF3 team decides to pass more than `.data` back at some point? That would force an API break.